product/updatePNCBuildConfigs.sh

```diff
@@ -37,51 +37,61 @@ initVariables() {
   # init variables used by both flows
   product_version=$DS_VERSION
   product_name="RHOSDS"
-  project_name="devspaces-server"
+  project_name="devspaces-server" # maps to project id = 1274
+  project_id="1274"
 }
 
 configureProductVersion() {
+  local product_version="$1"
   product_id=$(pnc product list --query "abbreviation==$product_name" | yq -r '.[].id')
-  product_id_version=$(pnc product list-versions "$product_id" | yq -r '.[] | select(.version == "'"$DS_VERSION"'") | .id')
+  product_id_version=$(pnc product list-versions "$product_id" | yq -r '.[] | select(.version == "'"$product_version"'") | .id')
   if [[ $product_id_version ]]; then
-    echo "[INFO] detected existing PNC version for $DS_VERSION, id - $product_id_version"
+    echo "[INFO] detected existing PNC version for $product_version: https://orch.psi.redhat.com/pnc-web/#/products/166/versions/${product_id_version}"
   else
-    echo "[INFO] creating PNC version for $DS_VERSION"
+    echo "[INFO] creating PNC version for $product_version"
     product_id_version=$(pnc product-version create --product-id="$product_id" "$product_version" | yq -r '.id')
-    echo "[INFO] creadted PNC version for $DS_VERSION, id - $product_id_version"
+    echo "[INFO] created PNC version for $product_version, id - $product_id_version"
   fi
 }
 
 configureLatestBuildConfig() {
+  local product_version="$1"
+  curl -sSLo /tmp/job-config.json https://raw.githubusercontent.com/redhat-developer/devspaces/devspaces-3-rhel-8/dependencies/job-config.json
   build_config_id=$(pnc build-config list --query "project.name==$project_name;productVersion.version==$product_version" | yq -r '.[].id')
```

Comment:
could this be the place, that caused a problem during the 3.6?
because next build config had 3.6 version in it, no latest build config was created?

Reply:
build_config_id=$(pnc build-config list --query "project.name==$project_name;productVersion.version==$product_version;name!=devspaces-server-build-main" | yq -r '.[].id')
```suggestion
  build_config_id=$(pnc build-config list --query "project.name==$project_name;productVersion.version==$product_version;name!=devspaces-server-build-main" | yq -r '.[].id')
```